python/dgl/backend/tensorflow/tensor.py

```diff
@@ -0,0 +1,564 @@
+    return tf.gather(data, row_index)
+
+
+def slice_axis(data, axis, begin, end):
+    # assert axis == 0
+    # tf doesn't behave well with negative
+    s = [slice(None) for i in range(data.ndim)]
+    if end == 0:
+        end = data.shape[axis]
+    s[axis] = slice(begin, end, None)
+    return data[tuple(s)]
+
+
+def take(data, indices, dim):
+    return tf.gather_nd(data, indices, dim)
+
+
+def narrow_row(x, start, stop):
+    return x[start:stop]
+
+
+def scatter_row(data, row_index, value):
+    row_index = tf.expand_dims(row_index, 1)
+    return tf.tensor_scatter_nd_update(data, row_index, value)
+
+
+def scatter_row_inplace(data, row_index, value):
+    raise NotImplementedError("Tensorflow doesn't support inplace update")
+
+
+def squeeze(input, dim):
+    return tf.squeeze(input, axis=dim)
+
+
+def unsqueeze(input, dim):
+    return tf.expand_dims(input, axis=dim)
+
+
+def reshape(input, shape):
+    return tf.reshape(input, shape)
+
+
+def swapaxes(input, axis1, axis2):
+    return tf.transpose(input, perm=[axis1, axis2])
+
+
+def zeros(shape, dtype, ctx):
+    with tf.device(ctx):
+        t = tf.zeros(shape, dtype=dtype)
+    return t
+
+
+def zeros_like(input):
+    return tf.zeros_like(input)
+
+
+def ones(shape, dtype, ctx):
+    with tf.device(ctx):
+        t = tf.ones(shape, dtype=dtype)
+    return t
+
+
+def uniform(shape, dtype, ctx, low, high):
+    with tf.device(ctx):
+        t = tf.random.uniform(shape, dtype=dtype, minval=low, maxval=high)
+    return t
+
+
+def pad_packed_tensor(input, lengths, value, l_min=None):
+    old_shape = input.shape
+    if isinstance(lengths, tf.Tensor):
+        max_len = as_scalar(lengths.max())
+    else:
+        max_len = builtins.max(lengths)
+
+    if l_min is not None:
+        max_len = builtins.max(max_len, l_min)
+
+    batch_size = len(lengths)
+    ndim = input.ndim
+    tensor_list = []
+    cum_row = 0
+    pad_nparray = np.zeros((ndim, 2), dtype=np.int32)
+    for l in lengths:
+        t = input[cum_row:cum_row+l]
+        pad_nparray[0, 1] = max_len - l
+        t = tf.pad(t, tf.constant(pad_nparray),
+                   mode='CONSTANT', constant_values=value)
+        tensor_list.append(t)
+        cum_row += l
+    return tf.stack(tensor_list, axis=0)
+
+
+def pack_padded_tensor(input, lengths):
+    out_list = []
+    for i, l in enumerate(lengths):
+        t = input[i]
+        out = t[:l]
+        out_list.append(out)
+    return tf.concat(out_list, axis=0)
+
+
+def unsorted_1d_segment_sum(input, seg_id, n_segs, dim):
+    assert dim == 0  # Why we need dim for 1d?
+    return tf.math.unsorted_segment_sum(input, seg_id, n_segs)
+
+
+def unsorted_1d_segment_mean(input, seg_id, n_segs, dim):
+    assert dim == 0  # Why we need dim for 1d?
+    return tf.math.unsorted_segment_mean(input, seg_id, n_segs)
+
+# TODO: TF has unsorted_segment_max, which can accelerate _max_on on batched graph
```

Comment:
@yzh119 Could you help review the four functions above? They are related to the glob module you wrote.